File: src/main/java/com/back/domain/bid/bid/dto/response/BidPageResponse.java
package com.back.domain.bid.bid.dto.response;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.data.domain.Page;

import java.util.List;

@Getter
@AllArgsConstructor
public class BidPageResponse {
    private List<BidListItemDto> content;
    private int page;
    private int size;
    private long totalElements;
    private int totalPages;

    public static BidPageResponse from(Page<BidListItemDto> page) {
        return new BidPageResponse(
                page.getContent(),
                page.getNumber(),
                page.getSize(),
                page.getTotalElements(),
                page.getTotalPages()
        );
    }
}

File: src/main/java/com/back/domain/bid/bid/dto/response/BidPageResponse.kt
package com.back.domain.bid.bid.dto.response

import org.springframework.data.domain.Page

class BidPageResponse(
    val content: List<BidListItemDto>,
    val page: Int,
    val size: Int,
    val totalElements: Long,
    val totalPages: Int
) {
    companion object {
        @JvmStatic
        fun from(page: Page<BidListItemDto>): BidPageResponse =
            BidPageResponse(
                content = page.content,
                page = page.number,
                size = page.size,
                totalElements = page.totalElements,
                totalPages = page.totalPages
            )
    }
}
